bin/remixd

```diff
@@ -1,46 +1,39 @@
 program
-.usage('-s <shared folder>')
-.description('Provide a two-way connection between the local computer and Remix IDE')
-.option('--remix-ide  <url>', 'URL of remix instance allowed to connect to this web sockect connection')
-.option('-s, --shared-folder <path>', 'Folder to share with Remix IDE')
-.option('--read-only', 'Treat shared folder as read-only (experimental)')
-.on('--help', function(){
-  console.log('\nExample:\n\n    remixd -s ./ --remix-ide http://localhost:8080')
-}).parse(process.argv)
+    .usage('[options]')
+    .description('Provide a two-way connection between the local computer and Remix IDE')
+    .option('--remix-ide  <url>', 'URL of remix instance allowed to connect to this web socket connection', 'https://remix.ethereum.org')
+    .option('-s, --shared-folder <path>', 'Folder to share with Remix IDE')
+    .option('--read-only', 'Treat shared folder as read-only (experimental)')
+    .option('--forward-commands', 'Enables CLI commands forwarding')
+    .option('--permissions <path>', 'What permissions are allowed')
+    .on('--help', function () {
+        console.log('\nExample:\n\n    remixd -s ./ --remix-ide http://localhost:8080\n')
+    }).parse(process.argv)
 
-var killCallBack = []
+if (program.sharedFolder) {
+    process.env.SHARED_FOLDER = program.sharedFolder;
+    console.log('\x1b[33m%s\x1b[0m', '[WARN] Any application that runs on your computer can potentially read from and write to all files in the directory.');
+    console.log('\x1b[33m%s\x1b[0m', '[WARN] Symbolic links are not forwarded to Remix IDE\n')
+}
 
-if (!program.remixIde) {
-  return console.log('\x1b[31m%s\x1b[0m', '[ERR] URL Remix IDE instance has to be provided.')
+if (program.readOnly) {
+    process.env.READ_ONLY = true;
 }
-console.log('\x1b[33m%s\x1b[0m', '[WARN] You may now only use IDE at ' + program.remixIde + ' to connect to that instance')
 
-if (program.sharedFolder) {
-  console.log('\x1b[33m%s\x1b[0m', '[WARN] Any application that runs on your computer can potentially read from and write to all files in the directory.')
-  console.log('\x1b[33m%s\x1b[0m', '[WARN] Symbolic links are not forwarded to Remix IDE\n')
-  var sharedFolderrouter = new Router(65520, servicesList['sharedfolder'], { remixIdeUrl: program.remixIde }, (webSocket) => {
-    servicesList['sharedfolder'].setWebSocket(webSocket)
-    servicesList['sharedfolder'].setupNotifications(program.sharedFolder)
-    servicesList['sharedfolder'].sharedFolder(program.sharedFolder, program.readOnly || false)
-  })
-  killCallBack.push(sharedFolderrouter.start())
+if (program.remixIde) {
+    origins.push(program.remixIde);
 }
 
-// kill
-function kill () {
-  for (var k in killCallBack) {
-    try {
-      killCallBack[k]()
-    } catch (e) {
-      console.log(e)
-    }
-  }
+if (program.permissions) {
+    process.env.PERMISSIONS = program.permissions;
 }
-process.on('SIGINT', kill) // catch ctrl-c
-process.on('SIGTERM', kill) // catch kill
-process.on('exit', kill)
```

Comment:
Why do we need custom SIGINT and SIGTERM handlers. Also SIGTERM is not supported on Windows. 
Basically killCallBack is just a list of running services that are then called with no arguments killCallBack[k]()

Reply:
Hmmm I don't really know... I think only @yann300  can help here.